### ISSUE TITLE
Collaborators needed!!

### ISSUE THREAD

**Issue by davidducos** [mydumper/mydumper#284]
Hello Community,

@maxbube and me are the only collaborators for MyDumper. we have several contributors, that we thank all the work done, but we need more commitment from a few of them. We reviewed all the issues and pull request and we found that @fredricj and @druud had been commenting and helping, here is the top10 list:
```
 156       "login": "maxbube",
  37       "login": "davidducos",
  18       "login": "fredricj",
  13       "login": "druud",
  12       "login": "utdrmac",
  12       "login": "UnixRonin",
  11       "login": "tsolodov",
   9       "login": "wagnerbianchijr",
   8       "login": "hydrapolic",
   7       "login": "nkadel-skyhook",
```
I'm creating this issues to invite them and others that want to participate. Please, comment on this if you want to and I will be discussing with Max. 

Thank you all!!

**Comment by verwilst** [comment]
I wonder why upstream doesn't pitch in a couple of hours each month instead of building mysqlpump. IMO mydumper is the best tool to take logical MySQL backups, hands down. Sad to see it so understaffed.

**Comment by VMEMachuca** [comment]
How could we collaborate to get support doe #237 ?

**Comment by verwilst** [comment]
I think people pushing PR's to this repository are "collaborating" as well, needing only a review from a maintainer to be included. This could be an awesome source for added features and bugfixes while taking a lot of work off your shoulders. It's a shame to see how many valid and sometimes awesome PR's are just sitting there for months/years untouched. 

So maybe you already have a lot of collaborators, they just need to get a bit of attention to get it into the codebase. ;-)

Just my 2 cents! (FWIW, I also added a PR just now, but when i noticed the backlog, my courage evaporated. :-) Hence this little message.)

**Comment by davidducos** [comment]
@verwilst your PR is already will be in release v0.10.5. I will be doing the review after publishing the release v0.10.3 which is schedule for next week. 
I'm trying to keep a calendar, releasing every 2 months. v0.10.1 was my first release with 'old' code, for the next release I started to review issues and PR. 
I used the word "collaborator", taking in account the github roles, for instance, if you open a ticket or ask for PR, you will be a contributor. A collaborator will be able to review/approve the PR and merge the code which is what we need, ah! and of course, it will need to test the code at least once! :D 
If you have another PR in the queue that I miss, please, ping me and I will be happy to review it.

**Comment by verwilst** [comment]
Awesome man, thanks for your work!

**Comment by SuperQ** [comment]
@davidducos I saw the Percona blog post, thanks for picking up work on this.

I'm not much of a C dev, I spend most of my time doing Go these days.

But, I was wondering if there would be interest in adding CI tests for PRs. I have a nice [CircleCI setup](https://github.com/prometheus/mysqld_exporter/blob/8e0d8568fa005ed2213feb424ec4e43ed8cca4e4/.circleci/config.yml#L70-L80) to do integration testing for various MySQL flavors. I could do a similar build/test framework here.

**Comment by davidducos** [comment]
Hi @SuperQ, any help is welcome! Could be possible for you to review code? this means, that one request to review, you download the code and test that it is doing whats suppose. 
About build/test, we are already have an environment that it is workin, but I'm open if we can develop something simpler! because, I would like to start building and testing with the latest version. Feel free to work on that project and let me know if you need my help!

**Comment by VMEMachuca** [comment]
We had recently up took v0.10.3 and confirm the benefit for out exports. We would love to take advantage of the new usingFastIndex functionality to improve our import times.

Could I get the code to help in any way?

**Comment by davidducos** [comment]
Hi @VMEMachuca,
If you want to use the fast index creation functionality, you are able to do so on this branch:
```
git clone https://github.com/maxbube/mydumper.git
cd mydumper
git checkout -t origin/usingFastIndex
```
It is not approved yet, and might not be packed in 0.10.5 if we don't have a review. Take into consideration that a bug was found and fixed already. Please, feel to test it and give your opinion on https://github.com/maxbube/mydumper/pull/286

Thanks!

**Comment by VMEMachuca** [comment]
Hi @davidducos,
I gave it a go with usingFastIndex and am getting some error messages:

`2021-04-27 04:17:10 [INFO] - Fast index creation will be use for table: 35k_020_cba_perfhw_3681_vm_mysql_3326.job_definition_snapshot
2021-04-27 04:17:10 [ERROR] - Error restoring 35k_020_cba_perfhw_3681_vm_mysql_3326.job_definition_snapshot from file 35k_cba_perfhw_3681_vm_20_mysql_3326.job_definition_snapshot-schema.sql.gz: Incorrect table definition; there can be only one auto column and it must be d
efined as a key
CREATE TABLE `job_definition_snapshot` (
  `seq` bigint(20) unsigned NOT NULL AUTO_INCREMENT,
  `hash` binary(20) NOT NULL,
  `iv` varchar(30) NOT NULL,
  `data` blob NOT NULL,
  PRIMARY KEY (`hash`)
) ENGINE=InnoDB DEFAULT CHARSET=utf8;`

**Comment by SuperQ** [comment]
@davidducos Is there a chat place we could collaborate? Maybe matrix.org `#mydumper`?

**Comment by davidducos** [comment]
Hi @SuperQ,
That is something that I thought too, but after discussing with @maxbube, we agreed that might not be useful (feel free to give your opinion about it and let's vote!)
So, first, we are not so many. After discussing in the chat we end up in an issue or pr... why don't we discuss directly on github?  and visibility is another reason, people outside of the matrix.org won't know about the discussions.

**Comment by SuperQ** [comment]
Sure, chatting in the open is better, which is why I suggested matrix.org (which can also bridge to old-school IRC like Freenode), not Slack. :grin: 

It was more of an idea for a faster, real-time, way to communicate. Threads on GitHub tend to get hard to read if you put a bunch of back and forth, support requests, etc. There's a new GitHub discussions feature, but I'm not convinced it's better.

**Comment by VMEMachuca** [comment]
Hi @davidducos,
Am also getting this error on two tables:
`
2021-04-27 05:26:44 [ERROR] - Error restoring 35k_020_cba_perfhw_3681_vm_mysql_3326.tabular_wid from file (null): Creating index 'IDX_WID' required more than 'innodb_online_alter_log_max_size' bytes of modification log. Please try again.
ALTER TABLE `35k_020_cba_perfhw_3681_vm_mysql_3326`.`tabular_wid`
 ADD  UNIQUE KEY `IDX_WID` (`wid`);

...

2021-04-27 05:59:54 [ERROR] - Error restoring 35k_020_cba_perfhw_3681_vm_mysql_3326.audited_instance from file (null): Creating index 'upd_txn' required more than 'innodb_online_alter_log_max_size' bytes of modification log. Please try again.
ALTER TABLE `35k_020_cba_perfhw_3681_vm_mysql_3326`.`audited_instance`
 ADD  KEY `upd_txn` (`upd_txn`);
2021-04-27 05:59:54 [INFO] - Thread 5 shutting down

...

**Comment by davidducos** [comment]
@VMEMachuca I created [this issue](https://github.com/maxbube/mydumper/issues/324) on your behalf.

**Comment by SuperQ** [comment]
@VMEMachuca Please do not post unrelated bug reports to this issue. If you find a problem, open a new issue, or comment on the related feature pull request.

**Comment by druud** [comment]
On 2021-04-27 14:05, VMEMachuca wrote:
> Hi @davidducos <https://github.com/davidducos>,
> I gave it a go with usingFastIndex and am getting some error messages:
> 
> |2021-04-27 04:17:10 [INFO] - Fast index creation will be use for table: 
> 35k_020_cba_perfhw_3681_vm_mysql_3326.job_definition_snapshot 2021-04-27 
> 04:17:10 [ERROR] - Error restoring 
> 35k_020_cba_perfhw_3681_vm_mysql_3326.job_definition_snapshot from file 
> 35k_cba_perfhw_3681_vm_20_mysql_3326.job_definition_snapshot-schema.sql.gz: 
> Incorrect table definition; there can be only one auto column and it 
> must be d efined as a key CREATE TABLE 
> |job_definition_snapshot|(|seq|bigint(20) unsigned NOT NULL 
> AUTO_INCREMENT,|hash|binary(20) NOT NULL,|iv|varchar(30) NOT 
> NULL,|data|blob NOT NULL, PRIMARY KEY (|hash|) ) ENGINE=InnoDB DEFAULT 
> CHARSET=utf8;|

Good catch, an autoinc can be done with an (even non-unique) secondary 
index, so that needs an exception.

Haven't tried yet to temporarily remove the autoinc as well.

Now I need to go and fix some internal library code ... :)

BTW, utf8mb4 is faster (and better tested) than utf8, good to alter that 
as well.

I still only do UTF-8-ness on the column level,
so all my tables have DEFAULT CHARSET=ascii (and I should maybe even 
switch to binary one day).
This to prevent having all textual columns be utf8mb4 by default,
as that would inflate indexes,
but I didn't benchmark that on MySQL 8 yet.

-- 
Greetings, Ruud

**Comment by davidducos** [comment]
Hi @druud, I already fix that branch, please #322
